fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -135,8 +140,11 @@
 
     private static final PropertyConverter propertyConverter = new PropertyConverter();
 
-    private static Function<FedoraResource, Function<IdentifierConverter<Resource, FedoraResource>, Function<Boolean,
-            Stream<Triple>>>> getDefaultTriples = resource -> translator -> uncheck(minimal -> {
+    // A curried type accepting resource, translator, and "minimality", returning triples.
+    private static interface RdfGenerator extends Function<FedoraResource,
```

Comment:
Well, it certainly reduces the repetitive cumbersome Function<<<>>> declarations, so it seems good to me.  If this pattern is useful in other implementations, it could be nice to make it part of the kernel-api.  But if it's not, then keeping it here and private makes sense to me.